crates/storage/src/connection/event.rs

```diff
@@ -34,6 +36,8 @@ pub struct EmittedEvent {
 pub enum EventFilterError {
     #[error("requested page size is too big, supported maximum is {0}")]
     PageSizeTooBig(usize),
+    #[error("event filter matches too many events")]
+    TooManyMatches,
```

Comment:
Should we add a suggestion for the user on how to resolve this? Poor example:
```suggestion
    #[error("Event query too broad. Reduce the block range or add more keys.")]
    TooManyMatches,
```